src/main/java/org/candlepin/insights/task/queue/TaskQueue.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.insights.task.queue;
+
+import org.candlepin.insights.task.TaskDescriptor;
+
+import org.springframework.beans.factory.DisposableBean;
+
+/**
+ * A TaskQueue is responsible for storing tasks until they are picked up by a registered processor
+ * for processing.
+ *
+ * @param <P> the type of TaskProcessor that will process the tasks in this queue.
+ */
+public interface TaskQueue<P extends TaskProcessor> extends DisposableBean {
```

Comment:
Rather than force TaskQueue implementations to implement DisposableBean, I'd prefer to just manually implement DisposableBean as an additional interface where necessary.

Reply:
Fix will be in next commit.